scapy/volatile.py

```diff
@@ -274,6 +274,115 @@ def __init__(self, template="*"):
+
+class GetMulticastMAC(RandMAC):
+	""" 
+	    ------------------------ 
+	    GetMulticastMAC(Index) - For use in fuzzing probes.
+	    --------------------------------------------------
+	    Get a multicast MAC from the list generated by IANA (http://www.iana.org/assignments/ethernet-numbers)
+		''' An Ethernet multicast address consists of the multicast bit, the
+		23-bit vendor component, and the 24-bit group identifier assigned by
+		the vendor.  For example, DEC is assigned the vendor component
+		08-00-2B, so multicast addresses assigned by DEC have the first
+		24-bits 09-00-2B (since the multicast bit is the low-order bit of the
+		first byte, which is "the first bit on the wire").'''
+		
+
+	    Index can be zero or another number.
+		- If valor is zero script will generate a random number (default Index = 0).
+		- Use index to generate a secuecial list. 
+	    Example:
+		...
+		>>> for i in range(66111): print GetMulticastMAC(i)
+		...
+		>>> GetMulticastMAC()
+		...
+	
+	+info: 
+		www.iniqua.com/labs
+	"""
+	def __init__(self, Index=0):
+		self.randMac = None
+		self.indexMac = None
+		self.mac = ()
+		template = "ff:ff:ff:ff:ff:ff"
+		if Index == 0:
+			rn = RandNum(0,66111)
+			if 0 <= rn < 63:
+ 				self.randMac = multicastList[rn]
+			elif 63 <= rn < 319:
+				template = multicastList[63]
+			elif 319 <= rn < 575:
+				template = multicastList[64]
+			elif 575 <= rn < 66111:
+				template = multicastList[65]
+                        template = template.split(":")
+                        for i in range(6):
+                	        if template[i] == "xx" and template[i-1] != "xx": 
+					v = RandByte()
+				elif template[i] == "xx" and template[i-1] == "xx":
+					v = v
+                      	        elif template [i] == "yy":
+					v = RandByte()
+				else:
+                                        v = int(template[i],16)
+                               	self.mac += (v,)
+			
+		else:
+			if 0 <= Index < 63:
+ 				self.indexMac = multicastList[Index]
+			if 63 <= Index < 319:
+				template = multicastList[63]
+				template = template.split(":")
+				relativeIndex = Index - 63
+				for i in range(6):
+					if template[i] == "xx":
+						v = int(hex(relativeIndex).split('x')[1],16)
+					else:
+						v = int(template[i],16)
+					self.mac += (v,)
+			
+			if 319 <= Index < 575:
+				template = multicastList[64]
+				template = template.split(":")
+				relativeIndex = Index - 319
+				for i in range(6):
+					if template[i] == "xx":
+						v = int(hex(relativeIndex).split('x')[1],16)
+					else:
+						v = int(template[i],16)
+					self.mac += (v,)
+				
+			if 575 <= Index < 66111:
+				template = multicastList[65]
+				relativeIndex = Index - 575
+				template = template.split(":")
+				#macof = hex(relativeIndex).split('x')[1]
+				macof = str("%04X" % relativeIndex)
+				if len(macof) < 4:
```

Comment:
that goes so deep :) if-if-if is mostly a good indication that you can move some functionality to a separate method(s).